### ISSUE TITLE
Possible bug: TypedReference* (and other restricted types) don't generate Warning CS8500

### ISSUE THREAD

**Issue by hamarb123** [dotnet/roslyn#65530]
**Version Used**: SDK 7.0.100 in VS 17.4.0

**Steps to Reproduce**:

When taking pointers to any managed type (classes, ref structs, and structs with object members), it is supposed to generate CS8500:
![image](https://user-images.githubusercontent.com/25316035/202929807-9a66884f-a0ca-4e28-89d0-0405b2cf5192.png)

These restricted types (`TypedReference`, `ArgIterator`, `RuntimeArgumentHandle`) are managed types and thus should generate the above warning.

Also, according to https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/low-level-struct-improvements.md#sunset-restricted-types, in the future these types may be viewed as ref structs, therefore they should give the warning now so users don't start randomly getting warnings.

```csharp
TypedReference* ptr = null;
```

**Expected Behavior**:

This should produce a warning like pointers to other managed types do.

**Actual Behavior**:

No warning (CS8500) is produced for these restricted types:
![image](https://user-images.githubusercontent.com/25316035/202929801-02d92823-8a08-4757-864b-d5555c0b8540.png)

Originally reported [here](https://github.com/dotnet/csharplang/discussions/6699)

**Comment by jcouv** [comment]
Those three restricted types are currently not considered managed types. This appears intentional (see test and code snippet below). The question is why? ;-)

```
        [Fact]
        public void IsManagedType_SpecialClrTypes()
        {
            var text = @"
class C { }
";
            var compilation = CreateCompilation(text);
            Assert.False(compilation.GetSpecialType(SpecialType.System_ArgIterator).IsManagedTypeNoUseSiteDiagnostics);
            Assert.False(compilation.GetSpecialType(SpecialType.System_RuntimeArgumentHandle).IsManagedTypeNoUseSiteDiagnostics);
            Assert.False(compilation.GetSpecialType(SpecialType.System_TypedReference).IsManagedTypeNoUseSiteDiagnostics);
        }
```

```
       private static (ThreeState isManaged, bool hasGenerics) IsManagedTypeHelper(NamedTypeSymbol type)
       {
...

           // Short-circuit common cases.
           switch (type.SpecialType)
           {
...
               case SpecialType.System_TypedReference:
               case SpecialType.System_ArgIterator:
               case SpecialType.System_RuntimeArgumentHandle:
                   return (ThreeState.False, false);
```

**Comment by hamarb123** [comment]
That's certainly interesting, not sure why that would have been done other than for the line of code I've attached for `TypedReference` from dotnet/runtime (since you couldn't do a pointer to it in the past obviously if it were marked as a managed type).
https://github.com/dotnet/runtime/blob/8d66372138fca4c9c73eef07c1632f13a0887d8e/src/libraries/System.Private.CoreLib/src/System/TypedReference.cs#L56
